lighthouse-core/test/audits/image-size-responsive-test.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * @license Copyright 2020 Sebastian Kreft All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const ImageSizeResponsiveAudit = require('../../audits/image-size-responsive.js');
+const assert = require('assert');
+
+/* eslint-env jest */
+
+const WIDTH = 800;
+const HEIGHT = 600;
+
+function generateImage(clientSize, naturalSize, props, src = 'https://google.com/logo.png') {
+  const image = {src, mimeType: 'image/png'};
+  const clientRect = {
+    clientRect: {
+      top: 0,
+      bottom: clientSize.displayedHeight,
+      left: 0,
+      right: clientSize.displayedWidth,
+    },
+  };
+  Object.assign(image, clientSize, naturalSize, clientRect, props);
+  return image;
+}
+
+describe('Images: size audit', () => {
+  function testImage(condition, data) {
+    const description = `identifies when an image ${condition}`;
+    it(description, () => {
+      const result = ImageSizeResponsiveAudit.audit({
+        ImageElements: [
+          generateImage(
+            {displayedWidth: data.clientSize[0], displayedHeight: data.clientSize[1]},
+            {naturalWidth: data.naturalSize[0], naturalHeight: data.naturalSize[1]},
+            data.props
+          ),
+        ],
+        ViewportDimensions: {
+          innerWidth: WIDTH,
+          innerHeight: HEIGHT,
+          devicePixelRatio: data.devicePixelRatio || 1,
+        },
+      });
+      let details = '';
+      if (result.score === 0) {
+        const {displayedSize: displayed, actualSize: actual, expectedSize: expected} =
+            result.details.items[0];
+        details = ` (displayed: ${displayed}, actual: ${actual}, expected: ${expected})`;
+      }
+      assert.strictEqual(result.score, data.score, `score does not match${details}`);
+    });
+  }
+
+  testImage('invalid image', {
+    score: 0,
+    clientSize: [100, 100],
+    naturalSize: [5, 5],
+  });
+
+  describe('is empty', () => {
+    testImage('is empty along width', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [0, 5],
+    });
+
+    testImage('is empty along height', {
+      score: 1,
+      clientSize: [100, 100],
+      naturalSize: [5, 0],
+    });
+  });
+
+  describe('too small', () => {
```

Comment:
```suggestion
  describe('too small to bother testing', () => {
```

is that right? I would expect these to not pass otherwise :)